src/application/main_menu/main_menu.c

```diff
@@ -0,0 +1,1525 @@
+        return FALSE;
+    }
+
+    *a2 = TRUE;
+
+    uVar5 = gSystem.touchY / 48;
+    if (data->unk138 == 0) {
+        if (uVar5 != 0) {
+            uVar5--;
+        }
+    } else if (data->unk138 == FX32_CONST(192)) {
+        uVar5 += 3;
+    } else {
+        uVar5 += 7;
+    }
+
+    if (uVar5 >= data->unk1B0) {
+        return FALSE;
+    }
+
+    data->unk54 = ov74_02227098(data, (u8)uVar5);
+
+    data->unk1AC = 1;
+    ov74_0222841C(data, data->unk54);
+    return ov74_022270C4(data, state, 0);
+}
+
+static BOOL ov74_02227428(MainMenuAppData *data, int *state) {
+    BOOL unk = FALSE;
+    BOOL unk2 = ov74_02227240(data, state, &unk);
+    if (unk) {
+        data->unk1AC = 1;
+        if (data->unk1B4 != 6) {
+            ov74_0222841C(data, 0xFF);
+        }
+        return unk2;
+    }
+    return ov74_02227180(data, state);
+}
+
+static u32 ov74_0222746C(MainMenuAppData *data, BOOL *a1) {
+    if (!System_GetTouchNew()) {
+        return 0;
+    }
+
+    int hitboxId = TouchscreenHitbox_FindRectAtTouchNew(sNewGameButtonHitboxes);
+    if (hitboxId == -1) {
+        return 0;
+    }
+
+    u16 something = 0xFFFE;
+    if (!DoesPixelAtScreenXYMatchPtrVal(data->bgConfig, GF_BG_LYR_MAIN_1, gSystem.touchX, gSystem.touchY, &something)) {
+        return 0;
+    }
+
+    u32 ret = 1;
+    *a1 = TRUE;
+    if (hitboxId != 0) {
+        ret = 2;
+    }
+
+    return ret;
+}
+
+static u32 ov74_022274D4(MainMenuAppData *data) {
+    BOOL local_10 = FALSE;
+    u32 uVar1 = ov74_0222746C(data, &local_10);
+    if (local_10) {
+        data->unk1AC = 1;
+        ov74_02228548(data, 0xFF);
+        return uVar1;
+    }
+
+    if (gSystem.newKeys == 0) {
+        return 0;
+    }
+
+    if (gSystem.newKeys & (PAD_BUTTON_Y | PAD_BUTTON_X | PAD_KEY_UP | PAD_KEY_DOWN | PAD_KEY_LEFT | PAD_KEY_RIGHT | PAD_BUTTON_B | PAD_BUTTON_A) && data->unk1AC == 1) {
+        data->unk1AC = 0;
+        ov74_02228548(data, data->unk56);
+        return 0;
+    }
+
+    if (gSystem.newKeys & (PAD_KEY_UP | PAD_KEY_DOWN)) {
+        data->unk56 ^= 1;
+        ov74_02228548(data, data->unk56);
+        PlaySE(SEQ_SE_DP_SELECT);
+        return 0;
+    }
+
+    u32 ret = 1;
+    if (gSystem.newKeys & PAD_BUTTON_A) {
+        if (data->unk56 != 0) {
+            ret = 2;
+        }
+    } else {
+        ret = 2;
+        if (!(gSystem.newKeys & PAD_BUTTON_B)) {
+            ret = 0;
+        }
+    }
+
+    return ret;
+}
+
+static u32 ov74_02227580(MainMenuAppData *data) {
+    return 0;
+}
+
+typedef struct {
+    u32 x;
+    u32 y;
+    u32 width;
+    u32 height;
+    u32 gmmId;
+    u32 msgId;
+    u32 unused18;
+} UnkStruct_ov74_0223BBD4;
+
+static BOOL ov74_02227584(MainMenuAppData *data) {
+    static UnkStruct_ov74_0223BBD4 ov74_0223BBD4[] = {
+        {
+         .x = 5,
+         .y = 5,
+         .width = 22,
+         .height = 14,
+         .gmmId = NARC_msg_msg_0800_bin,
```

Comment:
Ideally whenever we use a message const, we should add a comment next to it with its contents (mostly so the code can be understood easier and better documented)